File: src/main/java/com/vaadin/example/search/Categories.java
package com.vaadin.example.search;

public enum Categories {
    CLINIC_RECOMMENDATIONS("Клиничесие рекомендации"),
    ALGORITHM("Алгоритм");

    private String label;

    private Categories(String label) {
        this.label = label;
    }

    public String getLabel() {
        return label;
    }
}

File: src/main/kotlin/com/vaadin/example/search/Categories.kt
package com.vaadin.example.search

enum class Categories private constructor(val label: String) {
    CLINIC_RECOMMENDATIONS("Клиничесие рекомендации"),
    ALGORITHM("Алгоритм");

    companion object {
        fun getByLabel(label : String) : Categories? {
            val it = values().iterator()
            while (it.hasNext()) {
                val value = it.next();
                if (value.label.equals(label))
                    return value;
            }
            return null;
        }
    }
}
